sensor_msgs/include/sensor_msgs/image_encodings.hpp

```diff
@@ -120,13 +122,25 @@ const std::regex cv_type_regex("(8|16|32|64)(U|S|F)C([0-9]*)");
 // Utility functions for inspecting an encoding string
 static inline bool isColor(const std::string & encoding)
 {
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#else
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
```

Comment:
For the sake of maintenance, can we scope down where the ifdef/pragmas are required?

I think we could use a function like:

```
static inline bool isPackedYUV(const std::string& encoding)
```

That contains the deprecation suppression, and then use that in the other functions?

Reply:
Do you also want to use `isPackedYUV` as a public function? Eventually, the deprecated encodings will be removed. If this is a functionality that is supposed to be exposed to a user and used often, then I can add a `isPackedYUV`. But doing this just for the more narrow scoping of the encodings is a maintenance overhead IMHO.